### PR TITLE
Windows frida support

### PR DESCRIPTION
**Pull request by s1341**
Supercedes #1298

### CODE REVIEW

File: libafl/src/executors/inprocess.rs [hunk 1]
```diff
@@ -1261,6 +1261,17 @@ pub mod windows_exception_handler {
             let exception_list = data.exceptions();
             if exception_list.contains(&code) {
                 log::error!("Crashed with {code}");
+                #[cfg(all(feature = "std"))]
```
Comment (domenukk) [inline libafl/src/executors/inprocess.rs:1264]:
```rust
error: unneeded sub `cfg` when there is only one condition
    --> libafl/src/executors/inprocess.rs:1264:23
     |
1264 |                 #[cfg(all(feature = "std"))]
     |                       ^^^^^^^^^^^^^^^^^^^^ help: try: `feature = "std"`
     |
     
```

**Comment by s1341** [comment]
I need something from frida gum before I can merge. And Ole has yet to get to it.

**Comment by s1341** [comment]
This is the fix I was waiting for: https://github.com/frida/frida-gum/pull/754

**Comment by domenukk** [comment]
Sweet so now we need to wait for updated bindings and then this can be merged?

**Comment by s1341** [comment]
I still have to test the new solution... but yeah. It's a step closer.

File: fuzzers/frida_libpng/Cargo.toml [hunk 2]
```diff
@@ -26,11 +26,17 @@ reqwest = { version = "0.11.4", features = ["blocking"] }
...
+libafl = { path = "../../libafl/", features = [ "std", "llmp_compression",
+    "llmp_bind_public", "frida_cli", "errors_backtrace" ] } #,  "llmp_small_maps", "llmp_debug"]}
 libafl_bolts = { path = "../../libafl_bolts/" }
 frida-gum = { version = "0.13.2", features = [ "auto-download", "event-sink", "invocation-listener"] }
 libafl_frida = { path = "../../libafl_frida", features = ["cmplog"] }
 libafl_targets = { path = "../../libafl_targets", features = ["sancov_cmplog"] }
 libloading = "0.7"
 mimalloc = { version = "*", default-features = false }
 color-backtrace = "0.5"
+<<<<<<< Updated upstream
```
Comment (domenukk) [inline fuzzers/frida_libpng/Cargo.toml:38]: This probably doesn't want to be here

**Comment by s1341** [comment]
I'm now encountering the frida/capstone issue when trying to build for windows. I think we will need to merge #1720 and #1723 before I can continue in any significant way.

**Comment by s1341** [comment]
Ok. merged main after #1720 and #1723 were merged and migrated the hook_rt to yaxpeax (at least for x86_64). Now I can continue to test.

**Comment by tokatoka** [comment]
Is this ready? what are left to be done?

**Comment by s1341** [comment]
Tests now running and passing on linux and windows.

I noticed that there is a bug with register printing in the ASAN output.... incorrect registers are shown.

File: libafl/src/executors/inprocess.rs (hunk 1 above)
Comment (domenukk) [reply libafl/src/executors/inprocess.rs:1312]: This is still here :)

Comment (s1341) [reply libafl/src/executors/inprocess.rs:1312]: Fixed.

**Comment by s1341** [comment]
Looks like we need another new feature in frida in order to progress with this PR. I'm working on it...

**Comment by tokatoka** [comment]
what's missing?

**Comment by s1341** [comment]
Still waiting for a fix in frida to land.

**Comment by domenukk** [comment]
image

File: fuzzers/frida_gdiplus/src/fuzzer.rs [hunk 3]
```diff
@@ -6,9 +6,16 @@
 //! going to make it compilable only for Windows, don't forget to modify the
 //! `scripts/test_fuzzer.sh` to opt-out this fuzzer from that test.
 
+#[cfg(unix)]
```
Comment (domenukk) [inline fuzzers/frida_gdiplus/src/fuzzer.rs:9]: Should we use another allocator on Windows then?

Comment (domenukk) [reply fuzzers/frida_gdiplus/src/fuzzer.rs:9]: Ah lol nevermind it's the line below

File: fuzzers/frida_libpng/src/fuzzer.rs [hunk 4]
```diff
@@ -50,7 +50,9 @@ static GLOBAL: MiMalloc = MiMalloc;
 
 /// The main fn, usually parsing parameters, and starting the fuzzer
 pub fn main() {
+    env_logger::init();
     color_backtrace::install();
+    log::info!("hello!");
```
Comment (domenukk) [inline fuzzers/frida_libpng/src/fuzzer.rs:55]: Think we can do without this

File: fuzzers/qemu_systemmode/src/fuzzer_classic.rs [hunk 5]
```diff
@@ -5,7 +5,7 @@ use std::{env, path::PathBuf, process};
 
 use libafl::{
     corpus::{Corpus, InMemoryCorpus, OnDiskCorpus},
-    events::{launcher::Launcher, EventConfig},
+    events::{launcher::Launcher, EventConfig, CTRL_C_EXIT},
```
Comment (domenukk) [inline fuzzers/qemu_systemmode/src/fuzzer_classic.rs:8]: Is this change the failing qemu thingy?

File: libafl/src/events/llmp.rs [hunk 6]
```diff
@@ -1584,7 +1584,6 @@ where
                 compiler_fence(Ordering::SeqCst);
 
                 if child_status == CTRL_C_EXIT || staterestorer.wants_to_exit() {
-                    // if ctrl-c is pressed, we end up in this branch
```
Comment (domenukk) [inline libafl/src/events/llmp.rs:1587]: Why remove this comment?

**Review by domenukk — APPROVED** [review]

File: fuzzers/qemu_systemmode/src/fuzzer_classic.rs (hunk 5 above)
Comment (s1341) [reply fuzzers/qemu_systemmode/src/fuzzer_classic.rs:8]: Yes.

File: libafl/src/events/llmp.rs (hunk 6 above)
Comment (s1341) [reply libafl/src/events/llmp.rs:1587]: Merge fail?

File: fuzzers/qemu_systemmode/src/fuzzer_classic.rs (hunk 5 above)
Comment (domenukk) [reply fuzzers/qemu_systemmode/src/fuzzer_classic.rs:8]: Then we should revert this file to main

Comment (s1341) [reply fuzzers/qemu_systemmode/src/fuzzer_classic.rs:8]: can you do that easily?

Comment (tokatoka) [reply fuzzers/qemu_systemmode/src/fuzzer_classic.rs:8]: git checkout fuzzer_classic.rs -- origin/main

Comment (domenukk) [reply fuzzers/qemu_systemmode/src/fuzzer_classic.rs:8]: I'd do it the other way around (not sure if both work? :D )

so
`git checkout main -- ./fuzzers/qemu_systemmode/src/fuzzer_classic.rs`

File: libafl/src/events/llmp.rs (hunk 6 above)
Comment (domenukk) [reply libafl/src/events/llmp.rs:1587]: Want to un-remove it?

Comment (domenukk) [reply libafl/src/events/llmp.rs]: Done

File: fuzzers/qemu_systemmode/src/fuzzer_classic.rs (hunk 5 above)
Comment (domenukk) [reply fuzzers/qemu_systemmode/src/fuzzer_classic.rs]: Done

File: fuzzers/frida_libpng/src/fuzzer.rs (hunk 4 above)
Comment (domenukk) [reply fuzzers/frida_libpng/src/fuzzer.rs]: Moved into `fuzz` and changed the message

**Comment by domenukk** [comment]
@s1341 from my PoV this looks good now, feel free to merge

File: libafl_frida/src/utils.rs [hunk 7]
```diff
@@ -176,9 +201,45 @@ pub fn writer_register(reg: RegSpec) -> X86Register {
...
+            );
+            Err(Error::illegal_state(
+                "Instruction did not diassemble properly",
+            ))
+        }
+    }
+}
+
+#[cfg(target_arch = "aarch64")]
+pub(crate) fn frida_to_cs(
```
Comment (domenukk) [inline libafl_frida/src/utils.rs:226]: This function is never used(?)

**Comment by domenukk** [comment]
Any reason to keep `frida_to_cs` for aarch64? It's not in use

**Pull request merged**